### ISSUE TITLE
ActiveAdminDatetimepicker::Base mattr_accessor :format defaults to nil

### ISSUE THREAD

**Issue by gbirchmeier** [activeadmin-plugins/active_admin_datetimepicker#20]
(version 0.2.0, rails 4.0.14, ruby 2.1.0, activeadmin 1.0.0.pre1)

If I don't explicitly set `ActiveAdminDatetimepicker::Base.format = "%Y-%m-%d %H:%M"` in an initializer, I get this exception when I submit my form or load the edit-object form:

```
no implicit conversion of nil into String

top of framework trace:
  active_admin_datetimepicker (0.2.0) lib/active_admin_datetimepicker/base.rb:31:in `strftime'
  active_admin_datetimepicker (0.2.0) lib/active_admin_datetimepicker/base.rb:31:in `input_value'
  active_admin_datetimepicker (0.2.0) lib/active_admin_datetimepicker/base.rb:24:in `input_html_options'
```

I traced this to `lib/active_admin_datetimepicker/base.rb` [line 31](https://github.com/activeadmin-plugins/active_admin_datetimepicker/blob/master/lib/active_admin_datetimepicker/base.rb#L31).  The variable `format` is nil.

It appears that the `mattr_accessor :format` block (same file) is supposed to set a default value for `format`, but my experience today is that it's not doing anything.

**UPDATE:**

(from #19) `mattr_accessor :default_datetime_picker_options` has the same defect.

**Comment by Fivell** [comment]
@gbirchmeier , looks like you're right about default value, can you create PR with fix ?

**Comment by gbirchmeier** [comment]
I will attempt it.

**Comment by gbirchmeier** [comment]
This is really frustrating, but I can't seem to reproduce this in a spec.  See the specs I wrote in PR #23.

I figured out that on a live AA website, the exception is being thrown on a `DateTimePickerInput` object, which somehow gets created with both `mattr_attribute`s as `nil`.  However, when I create one of these objects in a spec, those values have non-nil values.

I don't know the innards of AA well enough to understand how it could create this object without giving those attrs a value.

@Fivell, do you have any idea?  I'm not sure how to approach this.

**Comment by Fivell** [comment]
@gbirchmeier ,  it is not obvious for me as well for now,  it will be very helpful if you can share some dummy application to let me reproduce it .

**Comment by gbirchmeier** [comment]
I have created a dummy rails app that replicates the bug.

Here you go: https://github.com/gbirchmeier/pickerbug

**Comment by Fivell** [comment]
@gbirchmeier , I figured out that default block for mattr_accessor can be used only from rails 4.1.0
you can see it by this commit https://github.com/rails/rails/commit/7dfbd91b0780fbd6a1dd9bfbc176e10894871d2d

So there  is 2 options, mention it in readme or setup default attributes manually. I think second option is ok, what do you think?

**Comment by Fivell** [comment]
fixed by #26

**Comment by gbirchmeier** [comment]
Ah, thanks.  That makes sense.

I've updated my demo project to 0.3.0 and it looks good now.  Problem solved.
